### PR TITLE
Adds "Highlights" to the Orbit menu

### PR DESCRIPTION
**Pull request by Contrabang**
## What Does This PR Do
Adds "Highlighting" to the observer's orbit menu, if a living player is followed by more than 20% of all observers they'll show up on this menu. The antagonist menu is shown above this panel, for both administrators and antagHUD users.
I'm considering a preference option for this, but I only want to implement it if someone would want this off. Also, very flexible with this PR, constructive criticism is appreciated.

It also adds a get_observers() helper, because apparently nothing like that exists?

## Why It's Good For The Game
 Its annoying to have to search through this list if you want to constantly keep an eye on two things (like a nuke disc and a nukie, or two antagonists demolishing security). This is a nice quality of life for people who want to be able to swap between both of them.

Now, there may be some concern about how this could be misconstrued as an antag hud. All this does is highlights what is already in the menu and puts it at the top if the bar. If someone was going to metagame this, they could do it already with just a bit more effort before. Plus, ghosts may be orbiting someone as a non-antag, but is doing something interesting (see that assistant that keeps running himself into that grille? Hilarious!)

## Images of changes

image

## Testing

Spawned skrell, put a blank mind in skrell, orbited skrell, opened menu, skrell was on the menu (delicious).

## Changelog
:cl:
add: A new "Highlights" section has been added to the observer orbit menu, showing what other fellow observers are looking at.
/:cl:

### CODE REVIEW

File: code/modules/mob/dead/observer/orbit.dm [hunk 1]
```diff
@@ -72,6 +73,8 @@
 			else if(M.stat == DEAD)
 				dead += list(serialized)
 			else
+				if(length(orbiters) >= 0.2 * length(get_observers())) // They're important if 20% of observers are watching them
```
Comment (lewcc) [inline code/modules/mob/dead/observer/orbit.dm:76]: This outer loop already iterates over pretty much every mob, please call this `get_observers()` function before the loop.

File: code/modules/mob/dead/observer/orbit.dm [hunk 2]
```diff
@@ -134,10 +138,14 @@
 					antag_serialized["antag"] = "Slaughter Demon"
 					antagonists += list(antag_serialized)
 		else
+			if(length(orbiters) >= 0.2 * length(get_observers())) // If a bunch of people are orbiting an object, like the nuke disk.
```
Comment (lewcc) [inline code/modules/mob/dead/observer/orbit.dm:141]: same here.

**Review by lewcc — CHANGES_REQUESTED** [review]
I like the idea, just needs fixing with implementation details. I feel like there should already be a proc that handles getting observers, but then again that's always the case with these helper functions

**Comment by lewcc** [comment]
One more thing, do you think you could sort this section by observer count? I think you might be able to do that on the js side, but it's been a while since I looked there.

**Comment by Contrabang** [comment]
Yeah, I tried that to no such luck, I'm no TGUI wizard yet. I tried some stuff with .sort and stuff, but with the info being transfered a list inside of a list which contains the number we want to search for, I couldnt figure it out.

**Review by Sirryan2002 — APPROVED** [review]
code looks good

**Review by SteelSlayer — APPROVED** [review]

**Comment by SteelSlayer** [comment]
Please rebuild TGUI to fix the conflict, then this can be merged.

**Comment by lewcc** [comment]
Looks good to me

**Review by lewcc — APPROVED** [review]

**Pull request merged**